packages/jest-worker/src/workers/WorkerAbstract.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {EventEmitter, PassThrough} from 'stream';
+import {
+  WorkerEvents,
+  WorkerInterface,
+  WorkerOptions,
+  WorkerStates,
+} from '../types';
+
+export default abstract class WorkerAbstract
+  extends EventEmitter
+  implements Pick<WorkerInterface, 'waitForWorkerReady' | 'state'>
```

Comment:
Create a new interface within `types.ts` instead of `Pick`ing here?

Reply:
It's an extremely limited use case just to prevent the class throwing typing errors, it didn't really seem necessary to create a new class just for that as it would have no utility elsewhere. Someone could create a different worker type in future and create a different abstract with totally different base props.